### PR TITLE
Add `update-versions` step

### PR DESCRIPTION
**Pull request by MikeRawding**
### Description
adds the [bump-versions](https://github.com/nicheinc/actions/pull/24/files#diff-2fb3ecbefe50ab9d3f65d3be8b17f6b5866297132d947d9b24f56fa392ab9331) step to the go build action.

### Testing Considerations

Here is a merge that triggered the new action.  Notice the merge commit message starts with `update-versions app patch chart patch`. https://github.com/nicheinc/gha-test-environment/commit/c2d2100e683faa75b6e6e80c38ab00dd570fc3db

That prompts the action to bump the versions like this https://github.com/nicheinc/gha-test-environment/commit/638cc5a0bce06d6b5b3c0cc53694b148340ddecf.  From there the flux deployment triggers as usual.

### Versioning

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204391427545928

### CODE REVIEW

File: .github/workflows/action.yaml [hunk 1]
```diff
@@ -63,6 +63,13 @@ on:
         required: true
 
 jobs:
+  update-versions:
+    name: Auto bump versions using commit messages
+    uses: nicheinc/actions/.github/workflows/bump-versions.yaml@feature/update-chart-release-description
```
Comment (jonathansharman) [inline .github/workflows/action.yaml:68]: 💭 This PR is based on `feature/copy-pr-description-to-release` anyway, and this probably goes without saying, but if we were to merge this into `v0`, we'd likely want to pin this branch to a commit rather than a branch so it still works if and when the branch is merged. (And eventually `bump-versions.yaml` should be moved to its own repo so we can pin to a specific version.)

File: .github/workflows/action.yaml [hunk 2]
```diff
@@ -136,6 +144,13 @@ jobs:
         service: ${{ fromJson(inputs.services) }}
     steps:
       - name: Checkout repo
+        if: ${{ needs.update-versions.outputs.versions-changed-by-ci == 'true' }}
+        uses: actions/checkout@v3
+        with:
+          ref: 'dev'
+      # No ref is important because the ref is set differently between a push and PR
```
Comment (jonathansharman) [inline .github/workflows/action.yaml:151]: ❓ Does `bump-versions` only do its thing on merge to `dev`? I'm having trouble understanding why we need to set the `ref` differently in these two cases.

File: .github/workflows/action.yaml [hunk 3]
```diff
@@ -521,13 +536,14 @@ jobs:
...
     if: ${{ github.event_name == 'push' || github.event_name == 'release' }}
-    uses: nicheinc/actions/.github/workflows/push-helm-charts.yaml@405c3f7cd949c0c557c85199d3206968916e5f6f
+    uses: nicheinc/actions/.github/workflows/push-helm-charts.yaml@feature/update-chart-release-description
     with:
       appchange: ${{ needs.build-and-ship-images.outputs.appchange }}
       appversion: ${{ needs.build-and-ship-images.outputs.appversion }}
       chartchange: ${{ needs.build-and-ship-images.outputs.chartchange }}
       chartversion: ${{ needs.build-and-ship-images.outputs.chartversion }}
       bypass-dev-protections: true
+      checkout-ref: 'dev'
```
Comment (jonathansharman) [inline .github/workflows/action.yaml:546]: ❓ Why is this necessary (now)?

File: .github/workflows/action.yaml (hunk 2 above)
Comment (MikeRawding) [reply .github/workflows/action.yaml:151]: @brittanysubialdea can be more help, she wrote this originally.  I think it's basically because `bump-versions` puts a commit on `dev` and `push-helm-charts` needs to check out `dev` to make sure it is up to date when it runs.

File: .github/workflows/action.yaml (hunk 1 above)
Comment (MikeRawding) [reply .github/workflows/action.yaml:68]: Yeah there is a little daisy chain of PRs.  I was thinking we just need to merge `actions`, then update `actions-go-ci` and merge that.

**Comment by MikeRawding** [comment]
Working after changing update-versions ref to be latest sha on master.  Test here

https://github.com/nicheinc/gha-test-environment/pull/37

File: .github/workflows/action.yaml (hunk 2 above)
Comment (brittanysubialdea) [reply .github/workflows/action.yaml:151]: ^yes its because both actions were committing separately that it needed the ref for 'dev' in particular. its been a while but there were weird errors with commit history if not for this

**Review by jonathansharman — APPROVED** [review]
